infra/build/functions/base_images.py

```diff
@@ -55,7 +57,40 @@ def _get_base_image_steps(images, tag_prefix=TAG_PREFIX):
             '-t',
             image,
             '-t',
-            f'{image}:{MAJOR_VERSION}',
+            f'{image}:{MAJOR_TAG}',
+            '.',
+        ],
+        'dir': 'oss-fuzz/infra/base-images/' + base_image,
+        'name': 'gcr.io/cloud-builders/docker',
+    })
+
+  return steps
+
+
+def _get_introspector_base_images_steps(images, tag_prefix=TAG_PREFIX):
+  """Returns build steps for given images version of introspector"""
+
+  steps = [{
+      'name':
+          'gcr.io/oss-fuzz-base/base-runner',
+      'args': [
+          'bash', '-c',
+          (f'sed -i s/base-clang/base-clang:{INTROSPECTOR_TAG}/g'
+           ' oss-fuzz/infra/base-images/base-builder/Dockerfile')
+      ]
+  }]
+
+  for base_image in images:
+    image = tag_prefix + base_image
+    build_arg = ''
+    if base_image == 'base-clang':
+        build_arg = 'introspector=1'
+    steps.append({
+        'args': [
+            'build',
+            '--build-arg': build_arg,
```

Comment:
Does passing --build-arg with an empty string work as expected (i.e. a no-op) ?

Reply:
You're right, the empty string breaks the build, did not test it properly last night!
Probably better approach would be passing build-arg for both image builds, in case of base-clang:introspector it is consumed. For the case of base-builder:introspector it produces a benign warning, but build succeeds.

Reply:
applied.